src/main.rs

```diff
@@ -150,80 +147,95 @@ pub async fn main() {
 	log::info!("Syncing block headers from 0 to {}", latest_block);
 
 	//tokio-tungesnite method for ws connection to substrate.
-	let url = url::Url::parse(&cfg.full_node_ws).unwrap();
-	let (ws_stream, _response) = connect_async(url).await.expect("Failed to connect");
+	let url = url::Url::parse(&cfg.full_node_ws)?;
+	let (ws_stream, _response) = connect_async(url).await?;
 	let (mut write, mut read) = ws_stream.split();
 
 	// attempt subscription to full node block mining stream
 	write
 		.send(Message::Text(
 			r#"{"id":1, "jsonrpc":"2.0", "method": "subscribe_newHead"}"#.to_string() + "\n",
 		))
-		.await
-		.unwrap();
+		.await?;
```

Comment:
Is there any error message we can show here?

Reply:
Added context here